wgpu-hal/src/vulkan/adapter.rs

```diff
@@ -798,6 +799,7 @@ impl super::InstanceShared {
                     || capabilities.supports_extension(vk::KhrDriverPropertiesFn::name());
 
                 let mut builder = vk::PhysicalDeviceProperties2KHR::builder();
+                builder = builder.push_next(&mut capabilities.maintenance_3);
```

Comment:
We should check if `driver_api_version >= 1.1` or if `VK_KHR_maintenance3` is supported.

We also need to optionally enable `VK_KHR_maintenance3` on vk < 1.1 in `get_required_extensions`.

Reply:
Modified as suggested

Reply:
Thanks! but we still need to optionally enable `vk::KhrMaintenance3Fn` on vk < 1.1 in `get_required_extensions`.